### PR TITLE
Catalog Client Sends Compressed Updates

### PR DESCRIPTION
**Pull request by nhazekam**
Compresses data for catalog updates using zlib.

### CODE REVIEW

File: dttools/src/catalog_query.c [hunk 1]
```diff
@@ -227,12 +229,28 @@ int catalog_query_send_update(const char *hosts, const char *text)
...
+	data_length = strlen(text)+1;
+
+	/* Estimates the bounds for the compressed data. */
+	compress_data_length = compressBound(data_length);
+	char* compress_data= malloc(compress_data_length);
+
+	/* Prefix the data with 0x1A (Control-Z) to indicate a compressed packet. */
+	compress_data[0] = 0x1A;
+
+	int success = compress((Bytef*)compress_data+1, &compress_data_length, (const Bytef*)text, data_length);
```
Comment (dthain) [inline dttools/src/catalog_query.c]: compress_data_length-1?

File: dttools/src/catalog_query.c [hunk 2]
```diff
@@ -227,12 +229,28 @@ int catalog_query_send_update(const char *hosts, const char *text)
...
+
+	if(success!=Z_OK) {
+		debug(D_DEBUG,"warning: Unable to compress data for update.\n");
+	}
```
Comment (dthain) [inline dttools/src/catalog_query.c]: If this fails, you will send garbage data.
Fall back by sending the data uncompressed..

File: dttools/src/catalog_query.c (hunk 1 above)
Comment (nhazekam) [reply dttools/src/catalog_query.c]: compress_data_length is an estimated value from compressBound and is then set by the compress function. I don't modify it here as I will modify it when we send it.

File: dttools/src/catalog_query.c (hunk 2 above)
Comment (nhazekam) [reply dttools/src/catalog_query.c]: Sounds good.

**Comment by dthain** [comment]
Now that I think about it, we should have a means of turning compression off in a pinch via an environment variable.  It's quite possible that someone will deploy new WQ code while relying on an old (local) catalog.  (We will update the global catalogs, of course.)

**Comment by nhazekam** [comment]
I'll add a env var to add this.

**Comment by nhazekam** [comment]
@dthain @btovar Please take a look. I can change the environment variable name and expected behavior if needed. Also, where should this be documented?

**Comment by dthain** [comment]
`doc/catalog.html`

**Comment by dthain** [comment]
Merge?

**Comment by nhazekam** [comment]
I am making a slight adjustment to the code since the compressed path of execution is no longer the default so we were doing excess work.

**Pull request merged**